### PR TITLE
Fix #3140: Instrumented tests are now shifted from unit test directory

### LINKED ISSUE #3140: Shift Instrumentation test written in Unit Test directory for AdministratorControlsFragmentTest
**Describe the bug**
There are some test cases written in `src/test/java/org/oppia/android/app/testing/administratorcontrols/AdministratorControlsFragmentTest.kt` which are instrumentation test but the file exist in unit test folder. (i.e. app/src/test)

**Expected behavior**
Look for the instrumentation tests in this file and shift them to the instrumentation test directory (i.e. app/src/sharedTest). 

**Additional context**
[How to test on Espresso?](https://developer.android.com/codelabs/advanced-android-kotlin-training-testing-basics#4)

**How to send a Pull Request for this Issue**
Add screenshot showing shifted tests are passing on Espresso.

### PR DESCRIPTION
**Pull request by jkugsiya**
## Explanation

Fixes #3140: Instrumented tests shifted from unit test directory 
## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

Screenshot of the passing UI tests that were shifted {Mobile Test(Physical device): Asus ROG Phone 3: API 29 }
Screenshot from 2021-05-12 13-01-09

Screenshot of the passing UI tests {Tablet (Emulator): Pixel C : API 28}
Screenshot from 2021-05-21 15-32-36

### CODE REVIEW

File: app/src/main/res/layout/app_version_fragment.xml
Comment (rt4914) [inline app/src/main/res/layout/app_version_fragment.xml]: Rename to `app_version_container` and `profile_list_container`

**Review by rt4914 — CHANGES_REQUESTED** [review]
@jkugsiya PTAL

File: app/src/test/java/org/oppia/android/app/testing/administratorcontrols/AdministratorControlsFragmentTest.kt
Comment (anandwana001) [inline app/src/test/java/org/oppia/android/app/testing/administratorcontrols/AdministratorControlsFragmentTest.kt:138]: this test should not be removed from here. as we are testing the fragment here, it's not a UI test.

Comment (anandwana001) [reply app/src/test/java/org/oppia/android/app/testing/administratorcontrols/AdministratorControlsFragmentTest.kt:138]: Though the similar test added in the new test file `testAdministratorControlsFragment_clickEditProfile_checkLoadingTheCorrectFragment` sounds good to me, but we need to check if that UI test is only working on a tablet or it needs to work on mobile too or not?

File: app/src/test/java/org/oppia/android/app/testing/administratorcontrols/AdministratorControlsFragmentTest.kt
Comment (anandwana001) [inline app/src/test/java/org/oppia/android/app/testing/administratorcontrols/AdministratorControlsFragmentTest.kt:158]: ditto

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt]: add named argument here

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt]: add named argument here

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt:137]: add named argument here

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt]: add an espresso result screenshot by running this test file on tablet configuration.

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt]:
```suggestion
    fun inject(administratorControlsFragmentTest: AdministratorControlsFragmentTest)
```

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt]:
```suggestion
    fun inject(administratorControlsFragmentTest: AdministratorControlsFragmentTest) {
```

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt]:
```suggestion
      component.inject(administratorControlsFragmentTest)
```

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt:107]: this test is similar to `testAdministratorControlsFragment_clickEditProfile_opensProfileListActivity` in the file `src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsActivityTest.kt`

We need a bit of extra work here to figure out what test should go under `AdministratorControlsActivityTest` and what test will go under `AdministratorControlsFragmentTest`. 

As I see in the `src/main/java/org/oppia/android/app/administratorcontrols/AdministratorControlsActivityPresenter.kt` we have functions like `loadAppVersion` and `loadProfileList` which loads the fragment, so the test which is testing the opening of the fragment will go under ActivityTest file and all the test which are testing the actions on the items which are inside the fragment will go under FragmentTest file.

Comment (anandwana001) [reply app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt:107]: Also take a look at `isMultipane` thing in `AdministratorControlsActivityPresenter` and check which tablet test will go in activity or fragment test file.

**Review by anandwana001 — CHANGES_REQUESTED** [review]
Thanks @jkugsiya 

Left few items to figure out.

File: app/src/main/res/layout/app_version_fragment.xml
Comment (jkugsiya) [reply app/src/main/res/layout/app_version_fragment.xml]: Renaming done from `app_version_fragment` to `app_version_container` but `profile_list_container `was already assigned to `profile_list_activity` so I used that only and removed id(s) from the `profile_list_fragment`: https://github.com/oppia/oppia-android/pull/3185/commits/586a89a87edeffeb1c2f2bedbe1f2c19ec255774

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt
Comment (jkugsiya) [reply app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt:137]: this is non-kotlin function! It gives error named-arguments are not allowed for non-kotlin functions.

File: app/src/test/java/org/oppia/android/app/testing/administratorcontrols/AdministratorControlsFragmentTest.kt
Comment (jkugsiya) [reply app/src/test/java/org/oppia/android/app/testing/administratorcontrols/AdministratorControlsFragmentTest.kt:138]: Can you please explain this a bit more: 

> but we need to check if that UI test is only working on a tablet or it needs to work on mobile too or not?

Comment (jkugsiya) [reply app/src/test/java/org/oppia/android/app/testing/administratorcontrols/AdministratorControlsFragmentTest.kt:138]: @anandwana001 I want to bring this into your notice: Running `testAdministratorControlsFragment_clickEditProfile_checkLoadingTheCorrectFragment` is failing as the fragment is already in view and after performing a click action over the `Edit Profiles` button it is loading another fragment on top of it instead it should check if the fragment is already in view then do not re-instantiate it again. The same is happening if we click more than one time on the `App Version` button. It is getting matched with multiple views because of re-instantiating the same fragment over the previously instantiated one.

**Comment by jkugsiya** [comment]
@anandwana001 I want to bring this into your notice for the tablet scenario: Running `testAdministratorControlsFragment_clickEditProfile_checkLoadingTheCorrectFragment` is failing as the fragment is already in view and after performing a click action over the `Edit Profiles` button it is loading another fragment on top of it instead it should check if the fragment is already in view then do not re-instantiate it again. The same is happening if we click more than one time on the `App Version` button. It is getting matched with multiple views because of re-instantiating the same fragment over the previously instantiated one.

Screenshots when double clicking `App Version` button:
image

Multiple views are getting instantiated:
image

File: app/src/main/java/org/oppia/android/app/administratorcontrols/AdministratorControlsActivityPresenter.kt
Comment (anandwana001) [inline app/src/main/java/org/oppia/android/app/administratorcontrols/AdministratorControlsActivityPresenter.kt]: why this change?

File: app/src/test/java/org/oppia/android/app/testing/administratorcontrols/AdministratorControlsFragmentTest.kt
Comment (anandwana001) [inline app/src/test/java/org/oppia/android/app/testing/administratorcontrols/AdministratorControlsFragmentTest.kt:104]: It looks like all the tests are removed from this file. 

As power this issue, what I meant by shifting is, we need to only move those tests which are instrumentation test, not the Unit test. 

Check my earlier review.

**Review by anandwana001 — CHANGES_REQUESTED** [review]

File: app/src/test/java/org/oppia/android/app/testing/administratorcontrols/AdministratorControlsFragmentTest.kt
Comment (jkugsiya) [reply app/src/test/java/org/oppia/android/app/testing/administratorcontrols/AdministratorControlsFragmentTest.kt:104]: Unit tests are now available in place: https://github.com/oppia/oppia-android/pull/3185/commits/5be55859c43b45f6e243ff4a95773fdeeda62ad8

File: app/src/main/java/org/oppia/android/app/administratorcontrols/AdministratorControlsActivityPresenter.kt
Comment (jkugsiya) [reply app/src/main/java/org/oppia/android/app/administratorcontrols/AdministratorControlsActivityPresenter.kt]: Change reverted: https://github.com/oppia/oppia-android/pull/3185/commits/5be55859c43b45f6e243ff4a95773fdeeda62ad8

File: app/src/test/java/org/oppia/android/app/testing/administratorcontrols/AdministratorControlsFragmentTest.kt
Comment (jkugsiya) [reply app/src/test/java/org/oppia/android/app/testing/administratorcontrols/AdministratorControlsFragmentTest.kt:138]: This test is now handled both for tablet and mobile.

File: app/src/test/java/org/oppia/android/app/testing/administratorcontrols/AdministratorControlsFragmentTest.kt
Comment (jkugsiya) [reply app/src/test/java/org/oppia/android/app/testing/administratorcontrols/AdministratorControlsFragmentTest.kt:158]: This test is now handled both for mobile and tablet.

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt
Comment (jkugsiya) [reply app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt]: Done adding named argument.

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt
Comment (jkugsiya) [reply app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt]: Done adding named argument.

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt
Comment (jkugsiya) [reply app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt]: Screenshot added in PR Explanation!

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt
Comment (jkugsiya) [reply app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt]: Done: https://github.com/oppia/oppia-android/pull/3185/commits/4f27d069811ce1d79a30f708609549c26518795b

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt
Comment (jkugsiya) [reply app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt]: Done https://github.com/oppia/oppia-android/pull/3185/commits/4f27d069811ce1d79a30f708609549c26518795b

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt
Comment (jkugsiya) [reply app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt]: Done https://github.com/oppia/oppia-android/pull/3185/commits/4f27d069811ce1d79a30f708609549c26518795b

File: app/src/main/java/org/oppia/android/app/administratorcontrols/AdministratorControlsActivityPresenter.kt
Comment (anandwana001) [inline app/src/main/java/org/oppia/android/app/administratorcontrols/AdministratorControlsActivityPresenter.kt:78]: why these changes?

**Review by anandwana001 — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/oppia/android/app/administratorcontrols/AdministratorControlsActivityPresenter.kt
Comment (anandwana001) [reply app/src/main/java/org/oppia/android/app/administratorcontrols/AdministratorControlsActivityPresenter.kt:78]: Also check this comment too - https://github.com/oppia/oppia-android/pull/3185#discussion_r631535910

Comment (jkugsiya) [reply app/src/main/java/org/oppia/android/app/administratorcontrols/AdministratorControlsActivityPresenter.kt:78]: I've done these changes because as I already mentioned that the application is loading the same fragment over the previously opened same fragmet if we click the button twice in multipane view. Also clarified this problem in this comment: https://github.com/oppia/oppia-android/pull/3185#issuecomment-841169003

**Comment by jkugsiya** [comment]
Any updates on this PR @prayutsu ??

**Review by prayutsu — APPROVED** [review]
LGTM

**Comment by jkugsiya** [comment]
PR is updated and is even with oppia:develop. The `PR don't merge - NEEDS UPDATE` label can be removed.https://github.com/oppia/oppia-android/pull/3185/commits/ab01adebcd7be57840f14758898120dcdd16b74a

**Comment by rt4914** [comment]
Will defer to @anandwana001 's review in this PR.

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt
Comment (anandwana001) [reply app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt:107]: @jkugsiya any update on this?

Comment (jkugsiya) [reply app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt:107]: Maybe we can remove the test `testAdministratorControlsFragment_clickEditProfile_checkSendingTheCorrectIntent` from the `AdministratorControlsFragmentTest` and let it be in the Activity test file.
I need some help if you don't think this should be done!

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt:147]: remote this part 
```
var fragment: Fragment? = null
      it.onActivity { activity ->
        fragment =
          activity.supportFragmentManager
            .findFragmentById(R.id.administrator_controls_fragment_multipane_placeholder)
      }
      assumeTrue(fragment == null)
```
Checking context in `sharedTest` is not a good idea, such things should be done only in unit test not the UI test.

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt:168]: remove this test

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt:153]:
```suggestion
  fun testAdministratorControlsFragment_clickEditProfile_profileListIsDisplayed() {
```

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt:162]: rather than testing whether a fragment is visible, let's test whether the recyclerview is visible or not and test whether the first item is admin or not. 
Test both these things in this test only.

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt:128]: As this screen is a multipane, so technically this will not send any intent, mean no new activity will be created. It's just a new fragment will appear on the right side. 

`testAdministratorControlsFragment_clickAppVersion_appVersionIsVisible`
I suggest removing `intended(hasComponent(AppVersionActivity::class.java.name))` and rather test if `app_version_text_view` is visible or not.

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt:91]: no need of intents, as we are not testing any activities.

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt:123]: Same applies here as of App version

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt:116]: remove 

```kt
var fragment: Fragment? = null
      it.onActivity { activity ->
        fragment =
          activity.supportFragmentManager
            .findFragmentById(R.id.administrator_controls_fragment_multipane_placeholder)
      }
      assumeTrue(fragment == null)
```

**Review by anandwana001 — CHANGES_REQUESTED** [review]

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt
Comment (anandwana001) [reply app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt:107]: Addressed on below comments. Let me know if a quick meeting is required for this PR.

**Comment by anandwana001** [comment]
@jkugsiya any update on this?

**Comment by jkugsiya** [comment]
> @jkugsiya any update on this?

We can have a meeting for me better understanding about the requirements. Apologies for the delay.

**Comment by anandwana001** [comment]
A meeting might be a bit tight scheduled these days, but I will try to find a slot and meet. By the time I suggest, if you looking forward to work on other issues, you can look for good first or good second issues. Also, comment on what issues you are facing in this PR.

**Comment by anandwana001** [comment]
Closing this PR as no further update.

**Pull request closed without merging**
